packages/material-ui-lab/src/internal/pickers/PickersToolbarText.tsx

```diff
@@ -1,46 +1,84 @@
 import * as React from 'react';
 import clsx from 'clsx';
 import Typography, { TypographyProps } from '@material-ui/core/Typography';
-import { MuiStyles, StyleRules, WithStyles, withStyles } from '@material-ui/core/styles';
+import { SxProps } from '@material-ui/system';
+import {
+  experimentalStyled,
+  unstable_useThemeProps as useThemeProps,
+  Theme,
+} from '@material-ui/core/styles';
+import {
+  unstable_composeClasses as composeClasses,
+  generateUtilityClass,
+  generateUtilityClasses,
+} from '@material-ui/unstyled';
 import { ExtendMui } from './typings/helpers';
 
 export interface PickersToolbarTextProps extends ExtendMui<TypographyProps> {
+  component?: React.ElementType;
```

Comment:
```suggestion
```

?

Reply:
without adding this, I got this type error.

![image](https://user-images.githubusercontent.com/18292247/116637614-9084bb00-a98e-11eb-9db1-309246a7e6bf.png)

At first I think that `Typography` should have prop `component` and it should pass to `PickersToolbarTextRoot` but not sure why it does not work so I add it in `PickersToolbarTextProps`

Reply:
`experimentalStyled` accepts additional props as a generic. Can't you use the Typography props there?

Reply:
No, this does not work.

```js
const PickersToolbarTextRoot = experimentalStyled<typeof Typography>(
  Typography,
  {},
  {
    name: 'MuiPickersToolbarText',
    slot: 'Root',
    overridesResolver: (props, styles) => styles.root,
  },
)(({ theme }) => ({
  transition: theme.transitions.create('color'),
  color: theme.palette.text.secondary,
  [`&.${pickersToolbarTextClasses.selected}`]: {
    color: theme.palette.text.primary,
  },
}));
```

Reply:
It looks a bit related to #13921